src/Workspaces/Core/Portable/Utilities/AsyncLazy`1.cs

```diff
@@ -294,6 +294,13 @@ private Request CreateNewRequest_NoLock()
 
         public override Task<T> GetValueAsync(CancellationToken cancellationToken)
         {
+            var cachedResult = _cachedResult;
+            if (cachedResult != null)
+            {
+                // Fastest possible path after the computation completes
+                return cachedResult;
+            }
```

Comment:
I will look into updating `GetValue` for consistency. Cancellation is not the fast path here so it would be second to a value already being available.